compiler/utils/astrepr.nim

```diff
@@ -248,115 +377,111 @@ proc treeRepr*(
 proc treeRepr*(
     conf: ConfigRef,
     pnode: PNode,
-    flags: set[TReprFlag] = defaultTreeReprFlags,
-    maxDepth: int          = 120,
-    maxLen: int            = 30,
-    maxPath: int           = 1,
-    indentIncrease: int = 0
+    rconf: TReprConf = defaultTReprConf,
+    indent: int = 0
   ): ColText =
   ## .. include:: tree_repr_doc.rst
 
-
   coloredResult(1)
 
 
-  var visited: Table[int, int]
-  var res = addr result
-  var nodecount = 0
+  let indentIncrease = indent
+  var
+    visited: Table[int, int]
+    res = addr result
+    nodecount = 0
+
   proc aux(n: PNode, idx: seq[int]) =
     var indent = indentIncrease
-    genFields(res[], indent, flags)
-
+    genFields(res[], indent, rconf)
     addIndent(indentIncrease)
-    if trfPositionIndexed in flags:
-      indent += 2
-      if 0 < idx.len:
-        var first = true
-        for idxIdx, pos in idx:
-          if idx.len - maxPath <= idxIdx:
-            indent += tern(first, 0, 1)
+    indent += 2
+    if 0 < idx.len:
+      var first = true
+      for idxIdx, pos in idx:
+        if idx.len - rconf.maxPath <= idxIdx:
+          indent += tern(first, 0, 1)
 
-            case pos:
-              of 0 .. 9: indent += 1
-              of 10 .. 99: indent += 2
-              else: indent += 3
+          case pos:
+            of 0 .. 9: indent += 1
```

Comment:
LoL, I tried to think of descriptors for these ranges, before giving up (don't think we need them) here is where I got to:

- we still got room
- little bit of spill over, this is fine
- look, just lock the door and no one will ever use that room again, it doesn't exist